modules/tinymce/src/themes/silver/main/ts/ui/core/ControlUtils.ts

```diff
@@ -22,9 +22,29 @@ const onSetupFormatToggle = (editor: Editor, name: string) => (api: Toolbar.Tool
   };
 
   // The editor may or may not have been setup yet, so check for that
-  editor.initialized ? init() : editor.on('init', init);
+  editor.initialized ? init() : editor.once('init', init);
 
-  return boundCallback.clear;
+  return () => {
+    editor.off('init', init);
+    boundCallback.clear();
+  };
+};
+
+const onSetupEvent = <T>(editor: Editor, event: string, f: (api: T) => void) => (api: T) => {
+  const handleEvent = () => f(api);
+
+  const init = () => {
+    f(api);
+    editor.on(event, handleEvent);
+  };
+
+  // The editor may or may not have been setup yet, so check for that
+  editor.initialized ? init() : editor.once('init', init);
+
+  return () => {
+    editor.off('init', init);
```

Comment:
Here also, do we need `.off` and `.once`? Or is it just a "to be safe" sorta thing